manifests/init.pp

```diff
@@ -24,8 +24,13 @@
   $repo_proxy_username                                 = $nodejs::params::repo_proxy_username,
   $repo_url_suffix                                     = $nodejs::params::repo_url_suffix,
   Array $use_flags                                     = $nodejs::params::use_flags,
+  String $darwin_package_provider                      = $nodejs::params::darwin_package_provider,
```

Comment:
Would it make sense to just name it ```package_provider``` and only override it in ```install.pp```? The other distros  can default to undef.

Reply:
Hmmm..maybe. Doing it the current way makes it more explicit that Darwin is it's own special snowflake rather than partially supporting your choice of `package_provider` regardless of os.

Reply:
I don't like special snowflakes :)

Reply:
> I don't like special snowflakes :)
Realistically noone is going to have any reason to change `package_provider` for any of the other operating systems supported by this module.

Reply:
Darwin is always its own special snowflake for anything puppet related. It always acts slightly differently (grumble... grump... etc.) We would need to pass the package_provider variable over to install and then use it there but it shouldn't change functionality in anyway - just makes it a bit harder to see where its going if you're using init. I don't mind one way or the other

Reply:
> Would it make sense to just name it package_provider and only override it in install.pp? The other distros can default to undef.

Apologies, I meant ```params.pp``` instead of ```install.p```.

> Realistically noone is going to have any reason to change package_provider for any of the other operating systems supported by this module.

From [pep 20](https://www.python.org/dev/peps/pep-0020/):

> Special cases aren't special enough to break the rules.

I don't see how renaming it to ```Optional[String] $package_provider``` and providing sane defaults in ```params.pp``` would be a difference for Darwin users but it would generalize it neatly.

Reply:
Have generalised this so passing in the package_provider for everything which can be overidden as and when!